src/components/Editor/index.js

```diff
@@ -160,16 +164,35 @@ class Editor extends Component {
     } else {
       this.undoTimestamp = 0
     }
+
+    if (evt.keyCode === 27 && !this.state.tabGuarded) {// Esc Key
+      this.setState({tabGuarded:true})
+    }else if(!(evt.shiftKey || evt.keyCode === 27 || evt.keyCode === 9) && this.state.tabGuarded){
```

Comment:
Is the behavior your'e seeing a shift press is being treated as hitting any old key (the assume desire to edit on any key press logic)? That would be expected based on how I was thinking of it, the evt.shiftKey was originally put in to handle shift tabbing not necessarily an independent shift press.